src/scalar.rs

```diff
@@ -400,6 +400,24 @@ impl Scalar {
         res
     }
 
+    /// Returns the bit representation of the given `Scalar` as
+    /// an array of 256 bits represented as `u8`.
+    pub fn to_bits(&self) -> [u8; 256] {
+        let bytes = self.to_bytes();
+        let mut res = [0u8; 256];
+
+        let mut j = 0;
+
+        for byte in &bytes {
+            for i in 0..8 {
+                let bit = byte >> i as u8;
+                res[j] = bit & 1u8;
+                j += 1;
+            }
+        }
+        res
+    }
```

Comment:
This won't take advantages of the auto vectorization (also, as personal preference I'd like avoiding having a "rogue" `j`).

Something like the following should be equivalent, and compiled in a more optimized / vectorized assembly:
```rust
     
     for (byte, bits) in bytes.iter().zip(res.chunks_mut(8)) {
         bits.iter_mut()
               .enumerate()
               .for_each(|(i, bit)| *bit = (byte >> i) & 1)
     }
     res
```

(Also the explicit casting should be not necessary, but double check)

Reply:
Yep, much better, just done. 

Thanks for pointing out!